kratos/processes/from_json_check_result_process.cpp

```diff
@@ -0,0 +1,866 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ \.
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Vicente Mataix Ferrandiz
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "containers/model.h"
+#include "processes/from_json_check_result_process.h"
+
+namespace Kratos
+{
+/// Local Flags
+KRATOS_CREATE_LOCAL_FLAG( FromJSONCheckResultProcess, CORRECT_RESULT,                 0 );
+KRATOS_CREATE_LOCAL_FLAG( FromJSONCheckResultProcess, HISTORICAL_VALUE,               1 );
+KRATOS_CREATE_LOCAL_FLAG( FromJSONCheckResultProcess, USE_NODE_COORDINATES,           2 );
+KRATOS_CREATE_LOCAL_FLAG( FromJSONCheckResultProcess, CHECK_ONLY_LOCAL_ENTITIES,      3 );
+KRATOS_CREATE_LOCAL_FLAG( FromJSONCheckResultProcess, NODES_CONTAINER_INITIALIZED,    4 );
+KRATOS_CREATE_LOCAL_FLAG( FromJSONCheckResultProcess, ELEMENTS_CONTAINER_INITIALIZED, 5 );
+KRATOS_CREATE_LOCAL_FLAG( FromJSONCheckResultProcess, NODES_DATABASE_INITIALIZED,     6 );
+KRATOS_CREATE_LOCAL_FLAG( FromJSONCheckResultProcess, ELEMENTS_DATABASE_INITIALIZED,  7 );
+
+/***********************************************************************************/
+/***********************************************************************************/
+
+FromJSONCheckResultProcess::FromJSONCheckResultProcess(
+    Model& rModel,
+    Parameters ThisParameters
+    ) : mrModelPart(GetModelPartFromModelAndSettings(rModel, ThisParameters)),
```

Comment:
This can be resolved using a Lambda function. I think this would make possible to remove the `GetModelPartFromModelAndSettings`.